Modules/@babylonjs/react-native/postinstall.js

```diff
@@ -1,13 +1,79 @@
 const os = require("os");
 const { spawn } = require('child_process');
 
+function iosCMake() {
+  const { spawn, spawnSync } = require('child_process');
+  const fs = require('fs');
+  const path = require('path');
 
+  // Determine cmake path in order of preference
+  let cmakePath = process.env.BABYLON_CMAKE_PATH;
+  
+  if (!cmakePath) {
+    // Check if cmake-runtime package is installed
+    try {
+      // Try multiple ways to find cmake-runtime
+      const possiblePaths = [
+        path.join(__dirname, 'node_modules', 'cmake-runtime', 'bin', 'cmake'),
+        path.join(process.cwd(), 'node_modules', 'cmake-runtime', 'bin', 'cmake'),
+        path.join(__dirname, '..', 'node_modules', 'cmake-runtime', 'bin', 'cmake'),
+        path.join(__dirname, '..', '..', 'node_modules', 'cmake-runtime', 'bin', 'cmake')
+      ];
+      
+      for (const possiblePath of possiblePaths) {
+        if (fs.existsSync(possiblePath)) {
+          cmakePath = possiblePath;
+          break;
+        }
+      }
+      
+      // If not found by path, try require.resolve as fallback
+      if (!cmakePath) {
+        try {
+          const cmakeRuntimePath = require.resolve('cmake-runtime/cmake');
+          if (fs.existsSync(cmakeRuntimePath)) {
+            cmakePath = cmakeRuntimePath;
+          }
+        } catch (e) {
+          // cmake-runtime not resolvable
+          console.warn(`cmake-runtime not resolvable.`);
+        }
+      }
+    } catch (e) {
+      // cmake-runtime not installed, continue to default
+      console.warn(`CMake not found in installed packages.`);
+    }
+  }
+  
+  if (!cmakePath) {
+    cmakePath = 'cmake';
+  }
 
-function iosCMake() {
+  // Check if cmake exists and get version
+  const versionResult = spawnSync(cmakePath, ['--version'], { encoding: 'utf8' });
```

Comment:
If we just force use the version we install from npm, then we also don't even need a version check. Seems like if we went that route this logic would really compress down into just a few lines of code.

Reply:
Indeed, it's way shorter but I don't get why it doesn't work sometimes: I create a folder. run `npm init` then `npm install path_to_tgz`.  cmake is not visible in `node_modules/.bin` and `npx cmake` errors. It seems to work correctly on CI though.